index.bs

```diff
@@ -8646,30 +8696,33 @@ a reference to the same object that the IDL value represents.
 </div>
 
 <div algorithm>
-    The <dfn export for="BufferSource">byte length</dfn> of a {{BufferSource}} |bufferSource| is
-    the value returned by the following steps:
+    The <dfn id=buffersource-byte-length export for="buffer source type">byte length</dfn> of a
+    [=buffer source type=] |bufferSource| is the value returned by the following steps:
 
     1.  Let |esBufferSource| be the result of [=converted to an ECMAScript value|converting=]
         |bufferSource| to an ECMAScript value.
     1.  If |esBufferSource| has a \[[ViewedArrayBuffer]] internal slot, then return
         |esBufferSource|.\[[ByteLength]].
     1.  Return |esBufferSource|.\[[ArrayBufferByteLength]].
 </div>
+<!-- TODO: should this be stricter around not being detached? -->
 
 <div algorithm>
-    The <dfn export for="BufferSource">underlying buffer</dfn> of a {{BufferSource}} |bufferSource|
-    is the value returned by the following steps:
+    The
+    <dfn id=buffersource-underlying-buffer export for="buffer source type">underlying buffer</dfn>
+    of a [=buffer source type=] |bufferSource| is the value returned by the following steps:
 
-    1.  If |bufferSource| is an {{ArrayBuffer}}, return |bufferSource|.
+    1.  If |bufferSource| is a [=buffer type=], then return |bufferSource|.
     1.  Let |esBufferSource| be the result of [=converted to an ECMAScript value|converting=]
         |bufferSource| to an ECMAScript value.
     1.  Return the result of [=converted to an IDL value|converting=]
-        |esBufferSource|.\[[ViewedArrayBuffer]] to an IDL value of type {{ArrayBuffer}}.
+        |esBufferSource|.\[[ViewedArrayBuffer]] to an IDL value of type [=buffer type=].
```

Comment:
This doesn't seem right. You need to switch on whether [[ViewedArrayBuffer]] IsSharedArrayBuffer() or not, and then convert appropriately.